frontend/src/components/Header.jsx

```diff
@@ -72,7 +89,10 @@ Header.propTypes = {
   request: requestProps.isRequired,
   train: trainProps,
   load: PropTypes.func.isRequired,
-  logout: PropTypes.func.isRequired
+  logout: PropTypes.func.isRequired,
+  setSearchQuery: PropTypes.func,
```

Comment:
if setSearchQuery is indeed always passed, set it to `isRequired`
Similar with `router` and `params` perhaps.

Again, I'm assuming this was to make tests pass. Based on that assumption: Logic here is similar to above. By lessening the restrictions here rather than following the restrictions in the test, we lose the development benefits of having stricter, more accurate prop types for the sake of tests.

Reply:
setSearchQuery has been set to isRequired